docs-src/cli/README.md

```diff
@@ -0,0 +1,234 @@
+
+## Installation
+
+Temporal CLI can be installed through several different methods. While most of them can be used across all operating systems, please note that Homebrew is macOS-exclusive.
+
+### cURL
+
+Run the following command to install the latest version of Temporal CLI.
+
+`curl -sSf https://temporal.download/cli.sh | sh`
+
+### Homebrew
+
+Run the following command in a new terminal window to install CLI for macOS.
+
+`brew install temporal`
+
+### GitHub releases
+
+Download and extract the latest release from [GitHub releases](https://github.com/temporalio/cli/releases).
+
+### CDN
+
+To install the Temporal CLI from CDN:
+
+1. Select the binary with the corresponding platform and architecture of your system.
+2. Download the binary.
+3. Extract the downloaded archive.
+4. Add the `temporal` binary to your system PATH.
+
+## Starting the Temporal Server
+
+Run the command provided below to start the Temporal Server.
+This will automatically start the Web UI.
+
+```bash
+temporal server start-dev
+```
+
+At this point you should have a server running on `localhost:7233` and a web interface at <http://localhost:8233>.
+A default Namespace has also been created.
+
+Run individual commands to interact with the local Temporal server.
+
+```bash
+temporal operator namespace list
+temporal workflow list
+```
+
+## Configuration
+
+Use the help flag to see a full list of CLI options:
+
+```bash
+temporal server start-dev --help
+```
+
+Configure the environment with `env` commands:
+
+```bash
+temporal env set [environment options]
+```
+
+### Namespace registration
+
+Namespaces are pre-registered at startup so they're available to use right away.
+To customize the pre-registered namespaces, start the server with:
+
+```bash
+temporal server start-dev --namespace foo --namespace bar
+```
+
+You can also register Namespaces with the following command:
+
+```bash
+temporal operator namespace create foo
+```
+
+### Persistence modes
+
+By default, `temporal server start-dev` runs in an in-memory mode.
+
+To persist the state to a file on disk, use `--db-filename`:
+
+```bash
+temporal server start-dev --db-filename my_test.db
+```
+
+### Enable or disable Temporal UI
+
+By default, the Temporal UI is started with Temporal CLI. The UI can be disabled via a runtime flag:
+
+```bash
+temporal server start-dev --headless
+```
+
+To build without static UI assets, use the `headless` build tag when running `go build`.
+
+<!--TODO: add go example -->
+
+### Dynamic configuration
+
+Advanced configuration of the Temporal CLI requires the use of a dynamic configuration file.
+This file is created outside of the Temporal CLI; it is usually located with the service's config files.
+
+Dynamic configuration values can also be set via `--dynamic-config-value KEY=JSON_VALUE`.
+For example, to disable the search attribute cache, run:
+
+```bash
+temporal server start-dev --dynamic-config-value system.forceSearchAttributesCacheRefreshOnRead=true
```

Comment:
This is actually already set by default, you might want to use a different dynamic config in the example.